### PR TITLE
Redis support

### PR DESCRIPTION
**Pull request by reinkrul**
TODO:

- [x] Waiting for https://github.com/nuts-foundation/go-stoabs/pull/13
- [x] Documentation
- [x] End to end test: https://github.com/nuts-foundation/nuts-go-e2e-test/pull/27

### CODE REVIEW

File: docs/pages/deployment/server_options.rst [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
+**Storage**                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
+storage.bbolt.backup.directory                                                                                                                                                                                                                                                                                                                      Target directory for BBolt database backups.
```
Comment (woutslakhorst) [inline docs/pages/deployment/server_options.rst:47]: The config has changed, this must be added to the changelog

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

**Review by woutslakhorst — APPROVED** [review]
We need a heads-up after merging. Config has changed for master.

**Comment by reinkrul** [comment]
> We need a heads-up after merging. Config has changed for master.

Wasn't entirely intentional, maybe we should revert it? On the other hand, I don't think BBolt backups were in v3 yet, so there's no reason we can't "break" it at this point since it wasn't in a release yet.

**Comment by reinkrul** [comment]
Please review docs

File: docs/pages/deployment/database-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Your node generates and stores private keys when you create DID documents or add new keys to it.
+Private keys are very sensitive, if you leak them others could alter your presence on the Nuts network and possibly worse.
+If you lose them you need to re-register your presence on the Nuts network, which could be very cumbersome.
+Thus, it's very important the private key storage is both secure and reliable.
+
+Filesystem
+==========
+
+This is the default backend but not recommended for production. It stores keys unencrypted on disk.
+Make sure to include the directory in your backups and keep these on a safe place.
```
Comment (woutslakhorst) [inline docs/pages/deployment/database-configuration.rst:54]:
```suggestion
Make sure to include the directory in your backups and keep these in a safe place.
```

File: docs/pages/deployment/database-configuration.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+=====
+
+This storage backend is the recommended way of storing secrets. It uses the `Vault KV version 1 store <https://www.vaultproject.io/docs/secrets/kv/kv-v1>`_.
+The prefix defaults to ``kv`` and can be configured using the ``crypto.vault.pathprefix`` option.
+There needs to be a KV Secrets Engine (v1) enabled under this prefix path.
+
+All private keys are stored under the path ``<prefix>/nuts-private-keys/*``.
+Each key is stored under the kid, resulting in a full key path like ``kv/nuts-private-keys/did:nuts:123#abc``.
+A Vault token must be provided by either configuring it using the config ``crypto.vault.token`` or setting the VAULT_TOKEN environment variable.
+The token must have a vault policy which enables READ and WRITES rights on the path. In addition it needs to READ the token information "auth/token/lookup-self" which should be part of the default policy.
```
Comment (woutslakhorst) [inline docs/pages/deployment/database-configuration.rst:67]:
```suggestion
The token must have a vault policy which has READ and WRITES rights on the path. In addition it needs to READ the token information "auth/token/lookup-self" which should be part of the default policy.
```

File: docs/pages/deployment/database-configuration.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+Migrating from to Vault
```
Comment (woutslakhorst) [inline docs/pages/deployment/database-configuration.rst:69]:
```suggestion
Migrating to Vault
```

File: docs/pages/deployment/database-configuration.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+=======================
+
+Migrating your private keys storage from the filesystem to Vault is relatively easy: just upload the keys to Vault under ``kv/nuts-private-keys``.
```
Comment (woutslakhorst) [inline docs/pages/deployment/database-configuration.rst:72]:
```suggestion
Migrating your private keys from the filesystem to Vault is relatively easy: just upload the keys to Vault under ``kv/nuts-private-keys``.
```

File: docs/pages/deployment/database-configuration.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Alternatively you can use the ``fs2vault`` crypto command, which takes the directory containing the private keys as argument (it assumes the container is called `nuts-node`):
```
Comment (woutslakhorst) [inline docs/pages/deployment/database-configuration.rst:74]:
```suggestion
Alternatively you can use the ``fs2vault`` crypto command, which takes the directory containing the private keys as argument (the example assumes the container is called `nuts-node`):
```

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

**Review by woutslakhorst — APPROVED** [review]
I won't complain about the missing copyright ;)

**Pull request merged**